nomad/state/paginator.go

```diff
@@ -88,7 +88,7 @@ func (p *Paginator) next() (interface{}, paginatorState) {
 	// have we found the token we're seeking (if any)?
 	id := raw.(IDGetter).GetID()
 	p.nextToken = id
-	if !p.nextTokenFound && id < p.seekingToken {
+	if !p.nextTokenFound && id != p.seekingToken {
```

Comment:
What happens if we the `X-Nomad-NextToken` is no longer valid at the time we make this call? Don't we end up skipping all the way to the end?

Reply:
Hum...that's right. I may have to go with my plan B then, which was to make `GetID` configurable to return either `ID` or `CreateIndex`. It does create a weird API though, because you sometimes paginate with an UUID and other times with a Raft index.

Back to the drawing board 😅

Reply:
Yeah I think it's the same problem either way, just sorting lexicographically vs numerically.